### ISSUE TITLE
CPM Train+Evaluation (Caffe+Matlab)

### ISSUE THREAD

**Issue by legan78** [ZheC/Realtime_Multi-Person_Pose_Estimation#74]
Dear Zhe Cao (@ZheC),

I trained various instances of your model using your caffe code and evaluated with your matlab code (4 scales). I have trained them with different configurations, i.e. different number of stages and different number of feature maps. Also I trained the complete model ( 6 stages with all feature maps as in the paper) trained until 440K iterations (same iterations as your provided model). Here I show you the graphs of the evaluation on COCO val2014 set I obtained using the matlab code.

![evaluation](https://user-images.githubusercontent.com/10438376/27596638-6fd6c8c4-5b60-11e7-8036-edbe4ad1cdc1.png)

As you can see in the yellow line (the 6 stage model with all feature maps) looks like the model is not going to converge to the same performance obtained with the model you provide (lines magenta and red). Moreover, the performance for models with less number of stages look very similar to the large one. 
My question are, does this behavior makes sense to you? Am I missing anything for the training (tunning, data, etc)?

I would appreciate your comments and suggestions regarding these results of training,
Regards

**Comment by ZheC** [comment]
Hey, the caffe code and training parameters are those I used to train the model. Do you follow the same code to create the LMDB for training? We are using both training and validation images from coco dataset to train the model, and only use around 2000 validation images for testing purpose, as in https://github.com/ZheC/Realtime_Multi-Person_Pose_Estimation/blob/master/training/genJSON.m#L36 

Overall, I feel it is strange that the 6 stages performance is about 3~4 percentage less accurate. From your graph, I also do not understand why 3 stages model get the same performance as the 6 stages model. When you trained the 6 stages model, what's the training loss of each stage?

**Comment by legan78** [comment]
Dear Zhe, 

Thank you for your answer. I have followed all the provided instructions for training. As for the training parameters I use the same solver configuration you provide in https://github.com/ZheC/Realtime_Multi-Person_Pose_Estimation/blob/master/training/example_proto/pose_solver.prototxt. I downloaded the ~189GB LMDB file you provide (instead of building it myself) and the evaluation have been performed over the small chunk of ~2000 images as in #36. 

Regarding to the models, it's worth to notice that I have trained from scratch (except the vgg part) each of them and not fine tuned from a trained network with one stage less. The overall loss for the network with 6 stages is about 600, and for each of the stages the loss is about 100.

Does this make sense?
Thanks a lot.

**Comment by legan78** [comment]
@ZheC So I was given a closer look to the json generated by genJSON.m. All possible ground truth is stored as well for the validation images. Do you perform any initialization on the images to be tested? e.g. only passing the ROI of the rectangle ground truth to the net to perform the detection?

**Comment by nicolocarissimi** [comment]
I'm interested in this issue too, since I will soon need to retrain the network from scratch.
Any news about it?

**Comment by aseaday** [comment]
Ｉhave met the same question.

**Comment by huaer9258** [comment]
Hello legan @legan78 , 

I got the same problems as yours. I also train the model using the same configurations as the owner provided and test it in matlab. But I get  nearly the same precision as yours (the yellow line). the average AP is nearly 52%. It has a good performance but there is still a distance from the author's 440k model whose precision is around 58%.  I'm verg confused. Do you know why does it happen? Have you get a progress now?

**Comment by YongboYu** [comment]
Hi, @legan78 @huaer9258 @ @aseaday , I met the same problem as you. Did you get the final results like the paper?
